core/primitives/src/shard_layout.rs

```diff
@@ -1353,4 +1392,93 @@ mod tests {
             )
         );
     }
+
+    #[test]
+    fn test_deriving_shard_layout() {
+        fn to_boundary_accounts<const N: usize>(accounts: [&str; N]) -> Vec<AccountId> {
+            accounts.into_iter().map(|a| a.parse().unwrap()).collect()
+        }
+
+        fn to_shard_ids<const N: usize>(ids: [u32; N]) -> Vec<ShardId> {
+            ids.into_iter().map(|id| ShardId::new(id as u64)).collect()
+        }
+
+        fn to_shards_split_map<const N: usize>(
+            xs: [(u32, Vec<u32>); N],
+        ) -> BTreeMap<ShardId, Vec<ShardId>> {
+            xs.into_iter()
+                .map(|(k, xs)| {
+                    (
+                        ShardId::new(k as u64),
+                        xs.into_iter().map(|x| ShardId::new(x as u64)).collect(),
+                    )
+                })
+                .collect()
+        }
```

Comment:
some of it may be already defined - see `new_shard_ids_vec`, `new_shards_split_map_v2`

Reply:
There're nicer for tests, fewer keystrokes by using arrays ;)) Do you think you would need these else where?